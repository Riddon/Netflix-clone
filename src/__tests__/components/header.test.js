import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Header } from "../../components/";
import { MemoryRouter } from 'react-router-dom';
import * as ROUTES from "../../constants/routes";
import logo from "../../logo.svg";

describe('<Header />', () => {
    it('renders the basic <Header/> with a background', () => {
        const{ container, getByText, getByAltText, getByTestId } = render(
            <MemoryRouter>
                <Header>
                    <Header.Frame>
                        <Header.Logo src="/logo.svg" alt="Netflix"/>
                        <Header.TextLink active="true">It is a link</Header.TextLink>
                    </Header.Frame>
                </Header>
            </MemoryRouter>
        );

        expect(getByTestId('header-bg')).toBeTruthy();
        expect(getByAltText('Netflix')).toBeTruthy();
        expect(getByText('It is a link')).toBeTruthy();

        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders the basic <Header/> without a background', () => {
        const{ container, getByText, getByAltText, queryByTestId } = render(
            <MemoryRouter>
                <Header bg={false}>
                    <Header.Frame>
                        <Header.Logo src="/logo.svg" alt="Netflix"/>
                        <Header.ButtonLink>Sign In</Header.ButtonLink>
                        <Header.TextLink active="true">It is a link</Header.TextLink>
                    </Header.Frame>
                </Header>
            </MemoryRouter>
        );

        expect(queryByTestId('header-bg')).toBeFalsy();
        expect(getByAltText('Netflix')).toBeTruthy();
        expect(getByText('It is a link')).toBeTruthy();
        expect(getByText('Sign In')).toBeTruthy();

        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders the full <Header/> with a background', () => {
        const { container, getByText, getByTestId } = render(
            <MemoryRouter>
                <Header src='joker1' dontShowOnSmallViewPort>
                    <Header.Frame>
                        <Header.Group>
                            <Header.Logo src='/images/logo.png' alt="Netflix"/>
                            <Header.TextLink
                                active='false'
                                onClick={() => {}}
                            >
                                Series
                            </Header.TextLink>
                            <Header.TextLink
                                active='true'
                                onClick={() => {}}
                            >
                                Films
                            </Header.TextLink>
                        </Header.Group>

                        <Header.Group>
                            <Header.Search searchTerm="Joker" setSearchTerm={()=> {}} />

                            <Header.Profile>
                                <Header.Picture src='/images/some.png'/>
                                <Header.Dropdown>
                                    <Header.Group>
                                        <Header.Picture src='/images/some.png'/>
                                        <Header.TextLink>Karlitto</Header.TextLink>
                                    </Header.Group>
                                    <Header.Group>
                                        <Header.TextLink onClick={() => {}}>Sign out</Header.TextLink>
                                    </Header.Group>
                                </Header.Dropdown>
                            </Header.Profile>
                        </Header.Group>
                    </Header.Frame>

                    <Header.Feature>
                        <Header.FeatureCallOut>Watch Joker Now</Header.FeatureCallOut>
                        <Header.Text>
                            Forever alone in a crowd
                        </Header.Text>
                        <Header.PlayButton>Play</Header.PlayButton>
                    </Header.Feature>
                </Header>
            </MemoryRouter>
        );

        expect(getByTestId('search-input')).toBeTruthy();
        expect(getByTestId('search-input').value).toBe('Joker');

        fireEvent.change(getByTestId('search-input'), {
           target: {value: 'Simpsons'},
        });
        fireEvent.click(getByTestId('search-click'));

        expect(getByText('Series')).toBeTruthy();
        expect(getByText('Films')).toBeTruthy();
        expect(getByText('Karlitto')).toBeTruthy();
        expect(getByText('Sign out')).toBeTruthy();
        expect(getByText('Watch Joker Now')).toBeTruthy();
        expect(getByText('Forever alone in a crowd')).toBeTruthy();
        expect(getByText('Play')).toBeTruthy();

        expect(container.firstChild).toMatchSnapshot();
    });
});


