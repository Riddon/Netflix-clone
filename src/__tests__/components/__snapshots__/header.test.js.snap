// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders the basic <Header/> with a background 1`] = `
<div
  class="header__Background-djltcn-0 bxpJXG"
  data-testid="header-bg"
>
  <div
    class="header__Container-djltcn-2 ckijbL"
  >
    <a
      href="/home"
    >
      <img
        alt="Netflix"
        class="header__Logo-djltcn-3 fvqKvo"
        src="/logo.svg"
      />
    </a>
    <p
      class="header__Link-djltcn-8 lkChII"
    >
      It is a link
    </p>
  </div>
</div>
`;

exports[`<Header /> renders the basic <Header/> without a background 1`] = `
<div
  class="header__Container-djltcn-2 ckijbL"
>
  <a
    href="/home"
  >
    <img
      alt="Netflix"
      class="header__Logo-djltcn-3 fvqKvo"
      src="/logo.svg"
    />
  </a>
  <a
    class="header__ButtonLink-djltcn-7 jxhhJx"
    href="/signin"
  >
    Sign In
  </a>
  <p
    class="header__Link-djltcn-8 lkChII"
  >
    It is a link
  </p>
</div>
`;

exports[`<Header /> renders the full <Header/> with a background 1`] = `
<div
  class="header__Background-djltcn-0 iNIYvk"
  data-testid="header-bg"
  src="joker1"
>
  <div
    class="header__Container-djltcn-2 ckijbL"
  >
    <div
      class="header__Group-djltcn-1 cIqXkZ"
    >
      <a
        href="/home"
      >
        <img
          alt="Netflix"
          class="header__Logo-djltcn-3 fvqKvo"
          src="/images/logo.png"
        />
      </a>
      <p
        class="header__Link-djltcn-8 iIwbJs"
      >
        Series
      </p>
      <p
        class="header__Link-djltcn-8 lkChII"
      >
        Films
      </p>
    </div>
    <div
      class="header__Group-djltcn-1 cIqXkZ"
    >
      <div
        class="header__Search-djltcn-4 hFcxWO"
      >
        <button
          class="header__SearchIcon-djltcn-6 efPWCS"
          data-testid="search-click"
        >
          <img
            alt="Search"
            src="/images/icons/search.png"
          />
        </button>
        <input
          class="header__SearchInput-djltcn-5 gAGPcF"
          data-testid="search-input"
          placeholder="Search films and series"
          value="Joker"
        />
      </div>
      <div
        class="header__Profile-djltcn-11 hRwWbG"
      >
        <button
          class="header__Picture-djltcn-9 eIbvSS"
          src="/images/users//images/some.png.png"
        />
        <div
          class="header__Dropdown-djltcn-10 jZOfLx"
        >
          <div
            class="header__Group-djltcn-1 cIqXkZ"
          >
            <button
              class="header__Picture-djltcn-9 eIbvSS"
              src="/images/users//images/some.png.png"
            />
            <p
              class="header__Link-djltcn-8 iIwbJs"
            >
              Karlitto
            </p>
          </div>
          <div
            class="header__Group-djltcn-1 cIqXkZ"
          >
            <p
              class="header__Link-djltcn-8 iIwbJs"
            >
              Sign out
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="header__Container-djltcn-2 header__Feature-djltcn-12 ckijbL cumYsh"
  >
    <h2
      class="header__FeatureCallOut-djltcn-14 egaGWq"
    >
      Watch Joker Now
    </h2>
    <p
      class="header__Text-djltcn-13 hSorVe"
    >
      Forever alone in a crowd
    </p>
    <button
      class="header__PlayButton-djltcn-15 enoKOF"
    >
      Play
    </button>
  </div>
</div>
`;
